core/src/main/scala/akka/persistence/cassandra/query/EventsByPersistenceIdStage.scala

```diff
@@ -69,39 +70,33 @@ import com.github.ghik.silencer.silent
       selectEventsByPersistenceIdQuery: PreparedStatement,
       selectSingleRowQuery: PreparedStatement,
       selectDeletedToQuery: PreparedStatement,
-      session: Session,
-      customConsistencyLevel: Option[ConsistencyLevel],
-      customRetryPolicy: Option[RetryPolicy]) {
+      session: CqlSession,
+      profile: String) {
 
     def selectEventsByPersistenceId(
         persistenceId: String,
         partitionNr: Long,
         progress: Long,
-        toSeqNr: Long,
-        fetchSize: Int)(implicit ec: ExecutionContext): Future[ResultSet] = {
+        toSeqNr: Long): Future[AsyncResultSet] = {
       val boundStatement =
-        selectEventsByPersistenceIdQuery.bind(persistenceId, partitionNr: JLong, progress: JLong, toSeqNr: JLong)
-      boundStatement.setFetchSize(fetchSize)
+        selectEventsByPersistenceIdQuery
+          .bind(persistenceId, partitionNr: JLong, progress: JLong, toSeqNr: JLong)
+          .setExecutionProfileName(profile)
       executeStatement(boundStatement)
     }
 
     def selectSingleRow(persistenceId: String, pnr: Long)(implicit ec: ExecutionContext): Future[Option[Row]] = {
-      val boundStatement = selectSingleRowQuery.bind(persistenceId, pnr: JLong)
-      session.executeAsync(boundStatement).asScala.map(rs => Option(rs.one()))
+      val boundStatement = selectSingleRowQuery.bind(persistenceId, pnr: JLong).setExecutionProfileName(profile)
```

Comment:
It is now important to set the right execution profile

We could have had a `CassandraSession` always use the same one but that would conflict with wanting to share the same CassandraSession for all parts of the plugin

Reply:
what settings will be used if profile is not set on the statement? I'm thinking about how it will be for the standalone Alpakka CassandraSession.

Reply:
All the driver's defaults which are defined in its reference.conf, I think that makes sense and not have our own defaults for a generic `CassandraSession`